### PR TITLE
Update JSON API for devise and doorkeeper's responses

### PR DESCRIPTION
**Pull request by hoangmirs**
## What happened
Update JSON API for devise and doorkeeper's responses

 
## Insight
- Customize the response of doorkeeper's endpoints by overriding body method
  + https://github.com/doorkeeper-gem/doorkeeper/wiki/Customizing-Token-Response
  + https://github.com/doorkeeper-gem/doorkeeper/wiki/Customizing-token-error-response
- Bring the message attribute to meta attribute
 

## Proof Of Work
- Forgot password endpoint:
```JSON
{
    "meta": {
        "message": "If your email address exists in our database, you will receive a password recovery link at your email address in a few minutes."
    }
}
```

- Doorkeeper's endpoints
```JSON
{
    "data": {
        "id": 27,
        "type": "token",
        "attributes": {
            "access_token": "WuE9Ows4HyyhujLUhNVewmTWQKI3kwLGgjkMa2q-ovY",
            "token_type": "Bearer",
            "expires_in": 7200,
            "refresh_token": "mPXOKuSlyGv3JjgQCBPNui7_Vk9bDjFp2yhqMyy-Z10",
            "created_at": 1596079132
        }
    }
}
```

### CODE REVIEW

File: lib/doorkeeper/custom_token_response.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Doorkeeper
+  module CustomTokenResponse
+    def body
+      {
+        data: {
+          id: @token.id,
+          type: :token,
```
Comment (olivierobert) [inline lib/doorkeeper/custom_token_response.rb:9]: Why is the `type` a symbol `:token` and not a string "token"?

File: app/controllers/api/v1/passwords_controller.rb [hunk 2]
```diff
@@ -8,7 +8,9 @@ class PasswordsController < Devise::PasswordsController
       protected
 
       def respond_with(*)
-        render json: { message: t('devise.passwords.send_paranoid_instructions') }
+        render json: {
+          meta: { message: t('devise.passwords.send_paranoid_instructions') }
+        }
```
Comment (olivierobert) [inline app/controllers/api/v1/passwords_controller.rb:13]: Seems a valid case for meta: https://jsonapi.org/format/#document-meta 👏

File: config/initializers/doorkeeper.rb [hunk 3]
```diff
@@ -494,3 +497,6 @@
   #
   # realm "Doorkeeper"
 end
+
+Doorkeeper::OAuth::TokenResponse.send :prepend, Doorkeeper::CustomTokenResponse
+Doorkeeper::OAuth::ErrorResponse.send :prepend, Doorkeeper::CustomErrorResponse
```
Comment (olivierobert) [inline config/initializers/doorkeeper.rb:502]: Next time, do think about links to the docs to document your PR e.g https://github.com/doorkeeper-gem/doorkeeper/wiki/Customizing-token-error-response

It helps reviewers in being confident in the implemented solution.

Comment (hoangmirs) [reply config/initializers/doorkeeper.rb:502]: Sure, next time I'll add more info, I thought that a short description is enough

File: lib/doorkeeper/custom_token_response.rb (hunk 1 above)
Comment (hoangmirs) [reply lib/doorkeeper/custom_token_response.rb:9]: I just prefer using symbol than string in this case, not sure there are any problems here.
I'll change back

Comment (olivierobert) [reply lib/doorkeeper/custom_token_response.rb]: Not sure if there is a problem either, just wondering why 😆

**Pull request merged**